Index: e16-0.16.8.12/config/enlightenment.menu
===================================================================
--- e16-0.16.8.12.orig/config/enlightenment.menu	2008-04-26 22:15:37.675004198 -0400
+++ e16-0.16.8.12/config/enlightenment.menu	2008-04-26 22:16:02.123005076 -0400
@@ -1,11 +1,12 @@
-"Enlightenment" "ROOT"
+"e16" "ROOT"
 "User Menus"		NULL menu file.menu
+"Debian Menus"          NULL menu /var/lib/e16/debian.menu
 "Desktop"		NULL menu desktop.menu
 "Settings"		NULL menu settings.menu
 "Themes"		NULL menu themes
 "Maintenance"		NULL menu maintenance.menu
 "Help"			NULL exec "edox $EROOT/E-docs"
-"About Enlightenment"	NULL about
+"About e16"     	NULL about
 "About this theme"	NULL exec "edox $ETHEME/ABOUT"
 "Restart"		NULL "exit restart"
 "Log Out"		NULL "exit logout"
Index: e16-0.16.8.12/misc/starte16
===================================================================
--- e16-0.16.8.12.orig/misc/starte16	2008-04-26 22:15:37.691004227 -0400
+++ e16-0.16.8.12/misc/starte16	2008-04-26 22:16:02.123005076 -0400
@@ -1,5 +1,9 @@
 #!/bin/sh
-case "$1" in
+
+session=$1
+test -z "$session" && session=${0##*starte16-}
+
+case "$session" in
 gnome|GNOME)
 	export WINDOW_MANAGER=e16
 	exec gnome-session
Index: e16-0.16.8.12/scripts/e_gen_menu
===================================================================
--- e16-0.16.8.12.orig/scripts/e_gen_menu	2008-04-26 22:15:37.703004301 -0400
+++ e16-0.16.8.12/scripts/e_gen_menu	2008-04-26 22:16:02.123005076 -0400
@@ -83,6 +83,7 @@
 @MainMenu = (
 	"t:User Menus",
 	"m:User Application List:user_apps.menu",
+	"m:Debian Menus:/var/lib/e16/debian.menu",
 	"m:GNOME:menus_GNOME/index.menu",
 	"m:KDE:menus_KDE/index.menu",
 	"m:Other:menus_Other/index.menu",
Index: e16-0.16.8.12/misc/e16-gnome.desktop.in
===================================================================
--- e16-0.16.8.12.orig/misc/e16-gnome.desktop.in	2008-04-26 22:15:37.679004251 -0400
+++ e16-0.16.8.12/misc/e16-gnome.desktop.in	2008-04-26 22:16:02.127004592 -0400
@@ -3,5 +3,5 @@
 Type=XSession
 Name=E16-GNOME
 Comment=This session starts GNOME using enlightenment (e16) as window manager
-Exec=e16-gnome
+Exec=/usr/share/e16/scripts/starte16-gnome
 Icon=@pkgdatadir@/misc/e16.png
Index: e16-0.16.8.12/misc/e16-kde.desktop.in
===================================================================
--- e16-0.16.8.12.orig/misc/e16-kde.desktop.in	2008-04-26 22:15:37.683004500 -0400
+++ e16-0.16.8.12/misc/e16-kde.desktop.in	2008-04-26 22:16:02.127004592 -0400
@@ -3,5 +3,5 @@
 Type=XSession
 Name=E16-KDE
 Comment=This session starts KDE using enlightenment (e16) as window manager
-Exec=e16-kde
+Exec=/usr/share/e16/scripts/starte16-kde
 Icon=@pkgdatadir@/misc/e16.png
Index: e16-0.16.8.12/misc/e16.desktop.in
===================================================================
--- e16-0.16.8.12.orig/misc/e16.desktop.in	2008-04-26 22:15:37.687004437 -0400
+++ e16-0.16.8.12/misc/e16.desktop.in	2008-04-26 22:16:02.127004592 -0400
@@ -3,5 +3,5 @@
 Type=XSession
 Name=E16
 Comment=This session starts the Enlightenment (e16) window manager
-Exec=e16
+Exec=/usr/share/e16/scripts/starte16
 Icon=@pkgdatadir@/misc/e16.png
